dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/task.py

```diff
@@ -170,6 +170,17 @@ def process_definition(self, process_definition: Optional[ProcessDefinition]):
         """Set attribute process_definition."""
         self._process_definition = process_definition
 
+    @property
+    def resource_list(self) -> List:
+        """Get task define attribute `resource_list`."""
+        resources = list()
+        for resource in self._resource_list:
+            if type(resource) == str:
+                resources.append(self.query_resource(resource))
+            elif type(resource) == dict and resource.get("id") is not None:
+                resources.append(resource)
```

Comment:
can we use `warnings.warn` to alert users this feat will is deprecation, should be use `List[str]` instead, and will be remove in version `3.2.0`?